ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/SampleManager.h

```diff
@@ -55,18 +62,27 @@ class SampleManager : public QObject
   Q_PROPERTY(double downloadProgress READ downloadProgress WRITE setDownloadProgress NOTIFY downloadProgressChanged)
   Q_PROPERTY(bool cancelDownload READ cancelDownload WRITE setCancelDownload NOTIFY cancelDownloadChanged)
   Q_PROPERTY(bool downloadFailed READ downloadFailed WRITE setDownloadFailed NOTIFY downloadFailedChanged)
+  Q_PROPERTY(QString api READ api CONSTANT)
+  Q_PROPERTY(Reachability reachability READ reachability NOTIFY reachabilityChanged)
 
 public:
   explicit SampleManager(QObject* parent = nullptr);
   ~SampleManager() override;
 
   Q_INVOKABLE virtual void init();
 
+  Q_INVOKABLE bool dataItemsExists();
+  Q_INVOKABLE void clearCredentialCache();
+  Q_INVOKABLE void downloadAllDataItems();
+  Q_INVOKABLE void downloadDataItemsCurrentSample();
+  Q_INVOKABLE bool deleteAllOfflineData();
   Q_INVOKABLE void setSourceCodeIndex(int i);
   Q_INVOKABLE void setupProxy(const QString& hostName, quint16 port, const QString& user, const QString& pw);
   Q_INVOKABLE void doneDownloading() { emit doneDownloadingChanged(); }
   Q_INVOKABLE void setApiKey(bool isSupportsApiKey = true);
 
+  SampleListModel* samples() const { return m_allSamples; }
```

Comment:
This could probably be private again. I think I had make it public when I was trying to access this in a derived class.